pkg/transport/handler.go

```diff
@@ -182,11 +180,8 @@ func HandleConnection(
 				netHandler.HandleResponse(targetConn, conn, netRequest, isInBoundConn, true)
 				closeConn(logger, targetConn)
 			}
-			wg.Add(1)
 			callCh <- respRoutine
 		}
-		wg.Wait()
-		close(callCh)
```

Comment:
A few clarifications: 
- control flow will never reach this point in code (it's either return from one of exit points inside the loop or keep pushing `respRoutine` inside `callCh` channel
- all exit points have `close` operation on `callCh`